ext: lib: mbedtls: Enable and set heap size at device startup

jukkar opened the pull request:
As the mbedtls heap is global for the whole device, enable it
during device startup if configured so. The heap size can be
set in config file.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
ext: lib: mbedtls: Enable and set heap size at device startup

As the mbedtls heap is global for the whole device, enable it
during device startup if configured so. The heap size can be
set in config file. There is no default value for the heap as
that depends very much on application needs.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

tbursztyka commented on ext/lib/crypto/mbedtls/Kconfig:
```diff
@@ -59,3 +59,24 @@ config MBEDTLS_INSTALL_PATH
+
+config MBEDTLS_HEAP_SIZE
+	int "Heap size for mbed TLS"
+	default 15000
```
That's super high, for most basic AES operation it's more like below 1Kb. High value is usually necessary for tls process, which needs to store keys etc...

You could set a small value by default, and tell in the help that depending on the usage of mtls it might need a large raise.

jukkar commented:
> That's super high, for most basic AES operation it's more like below 1Kb. High value is usually necessary for tls process, which needs to store keys etc...

This really depends on usage, for normal server application, the usage is typically (at least in my tests) between 10-15 kb.

> 
> You could set a small value by default, and tell in the help that depending on the usage of mtls it might need a large raise.

If we have too small value, then mbedtls will fail mysteriously in various places. I spent whole day last week debugging a weird connectivity issue where the fix was to increase the heap size.

Perhaps it is then better to leave the default value away, and make the compilation to fail if user has not set the value. That way we do not accidentally use too small value.

jukkar committed:
net: http: Remove mbedtls heap setting from http library

The global mbedtls heap is set automatically now so no need to
set it individually in the http library.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

tbursztyka reviewed (approved):
Yep, better let the user setting this value. It's very much dependent on use case indeed.

jukkar commented:
@galak / @nashif Are you ok with this change?
